palantiri/core/datahandler.py

```diff
@@ -88,21 +89,22 @@ def __repr__(self):
 
     def dump(self, to_insert):
         cur = self.conn.cursor()
-        cur.executemany(("INSERT INTO page(url, content, datescraped, status_code, headers) "
-                         "VALUES (%s, %s, %s, %s, %s) "
-                         "ON CONFLICT (url) DO NOTHING;"),
+        logging.info("************* Saving %d pages to DB" % len(to_insert))
+        cur.executemany(("INSERT INTO page(url, content, datescraped, CrawlerId) "
+                         "VALUES (%s, %s, %s, %s) ;"),
                         to_insert)
         cur.close()
         self.conn.commit()
 
     def find_by_id(self, _id, attempt = 0):
-        try:
-            cur = self.conn.cursor()
-            cur.execute(
-            """SELECT id FROM page WHERE url = '{}'""".format(_id)
-                )
-            return cur.fetchone() is not None
-        except (psycopg2.IntegrityError, psycopg2.InternalError):
-            if attempt < 5:
-                return self.find_by_id(_id, attempt + 1)
-            return None
+        with self.conn.cursor() as cur:
```

Comment:
Ugh... Guess I didn't close the cursor. I think we should discuss removing this function. It may be simpler and maybe even faster or more reliable to just rely on a `ON CONFLICT` with an `UPDATE content ...`.

Reply:
Yeah, I'd have to double check but I don't know where we are using this right now.